packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,197 @@
+            new Container(
+                decoration:
+                    const BoxDecoration(color: CupertinoColors.activeBlue),
+                constraints:
+                    const BoxConstraints(maxWidth: 2.0, minWidth: 2.0)),
+            widget.options.elementAt(i).key, (Key state) {
+          setState(() {
+            sharedState = state;
+          });
+        }, sharedState, true, false, false));
+      }
+    }
+
+    /// Returns a [Container] visually styled like the iOS segmented control bar
+    return Container(
+      margin: _kHorizontalItemPadding,
+      child: new Align(
+        child: new Container(
+          decoration: const ShapeDecoration(
+            color: CupertinoColors.white,
+            shape: const RoundedRectangleBorder(
+              side: BorderSide(
+                  color: CupertinoColors.activeBlue,
+                  width: 2.0,
+                  style: BorderStyle.solid),
+              borderRadius: BorderRadius.all(defaultCornerRadius),
+            ),
+          ),
+          child: new Row(
+              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+              children: buttons),
+        ),
+      ),
+    );
+  }
+}
+
+class _SegmentedControlButton extends StatefulWidget {
+  _SegmentedControlButton(this.selectState, this.widgetKey, this.valueChanged,
+      this.sharedState, this.isDivider, this.isFirst, this.isLast);
+
+  final Widget selectState;
+  final Key widgetKey;
+  final ValueChanged<Key> valueChanged;
+  Key sharedState;
+  final bool isDivider;
+  final bool isFirst;
+  final bool isLast;
+
+  @override
+  _SegmentedControlButtonState createState() => _SegmentedControlButtonState();
+}
+
+class _SegmentedControlButtonState extends State<_SegmentedControlButton> {
+  static const Radius defaultInnerCornerRadius = Radius.circular(1.0);
+  static const Color onPressColor = const Color(0x33007aff);
+  bool isPressed = false;
+
+  _SegmentedControlButtonState();
+
+  static const BorderRadius leftmostWidgetBorder = const BorderRadius.only(
+      topLeft: defaultInnerCornerRadius, bottomLeft: defaultInnerCornerRadius);
+  static const BorderRadius rightmostWidgetBorder = const BorderRadius.only(
+      topRight: defaultInnerCornerRadius,
+      bottomRight: defaultInnerCornerRadius);
+
+  @override
+  Widget build(BuildContext context) {
+    final bool isSelected = widget.widgetKey == widget.sharedState;
+    final TextStyle textStyle = DefaultTextStyle.of(context).style.copyWith(
+          color:
+              isSelected ? CupertinoColors.white : CupertinoColors.activeBlue,
+        );
+    final IconThemeData iconTheme = IconThemeData(
+      color: isSelected ? CupertinoColors.white : CupertinoColors.activeBlue,
+    );
+
+    Widget button = new Container(
+        constraints: const BoxConstraints(maxHeight: 28.0),
+        decoration: ShapeDecoration(
+          color: widget.isDivider
+              ? CupertinoColors.activeBlue
```

Comment:
You might move this complex conditional out, and compute it before you get in here, just for readability.